packages/devtools_app/lib/src/service_manager.dart

```diff
@@ -137,6 +137,8 @@ class ServiceConnectionManager {
 
     connectedApp = ConnectedApp();
 
+    unawaited(onClosed.then((_) => vmServiceClosed()));
```

Comment:
lesson learned: be careful to do all work that could possibly be done before an await before the await to avoid deadlock bugs like this.